style.lua

```diff
@@ -119,14 +144,31 @@ function filter_tags_way (keyvalues, numberofkeys)
       return filter, keyvalues, polygon, roads
    end
 
-  -- Treat objects with a key in polygon_keys as polygon
+   -- Treat objects with a key in polygon_keys as polygon
    for i,k in ipairs(polygon_keys) do
       if keyvalues[k] then
+         polygontag = 1
+         -- However, if the key/value combination occurs in linestring_values, do not treat the object as polygon
+         for index,tag in pairs(linestring_values) do
+            if k == tag[1] and keyvalues[k] == tag[2] then
+               polygontag = 0
+               break
+            end
+         end
+         if polygontag == 1 then
+            polygon = 1
```

Comment:
I believe you forgot a  break here. Further, I don't quite see what is the use of 'polygontag'. Setting 'polygon' directly should have the same effect.